torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4547,6 +4547,22 @@ def _tensor(shape, dtype=dtype, low=None, high=None):
 
     return [SampleInput(tensor, args=args) for tensor, args in test_cases]
 
+def sample_inputs_mse_loss(op_info, device, dtype, requires_grad, **kwargs):
+    def make_tensor_(shape):
+        return make_tensor(shape, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    sample_inputs = [
+        SampleInput(make_tensor_(shape), args=(make_tensor_(shape),)) for shape in ((), (S,), (M, S), (L, M, S))
```

Comment:
Yikes that's a big sample input! Maybe we could make it smaller so the tests don't take too long? (Do they take a long time?)

Reply:
Quick benchmark: running `pytest test/test_ops.py -k mse_loss --durations=5`

1. "as is"

```
0.71s call     test/test_ops.py::TestGradientsCUDA::test_fn_gradgrad_nn_functional_mse_loss_cuda_float64
0.43s call     test/test_ops.py::TestGradientsCUDA::test_fn_grad_nn_functional_mse_loss_cuda_float64
0.36s call     test/test_ops.py::TestCommonCUDA::test_dtypes_nn_functional_mse_loss_cuda
0.26s call     test/test_ops.py::TestMathBitsCUDA::test_neg_view_nn_functional_mse_loss_cuda_float64
0.25s call     test/test_ops.py::TestGradientsCUDA::test_forward_mode_AD_nn_functional_mse_loss_cuda_float64
```

2. replace `M` and `L` with `S`

```
0.70s call     test/test_ops.py::TestGradientsCUDA::test_fn_gradgrad_nn_functional_mse_loss_cuda_float64
0.43s call     test/test_ops.py::TestGradientsCUDA::test_fn_grad_nn_functional_mse_loss_cuda_float64
0.36s call     test/test_ops.py::TestCommonCUDA::test_dtypes_nn_functional_mse_loss_cuda
0.27s call     test/test_ops.py::TestMathBitsCUDA::test_neg_view_nn_functional_mse_loss_cuda_float64
0.26s call     test/test_ops.py::TestGradientsCUDA::test_forward_mode_AD_nn_functional_mse_loss_cuda_float64
```

3. remove inputs with more than 1 dimension (and replace them with scalar inputs to have a constant number of sample inputs)

```
0.69s call     test/test_ops.py::TestGradientsCUDA::test_fn_gradgrad_nn_functional_mse_loss_cuda_float64
0.44s call     test/test_ops.py::TestGradientsCUDA::test_fn_grad_nn_functional_mse_loss_cuda_float64
0.37s call     test/test_ops.py::TestCommonCUDA::test_dtypes_nn_functional_mse_loss_cuda
0.27s call     test/test_ops.py::TestMathBitsCUDA::test_neg_view_nn_functional_mse_loss_cuda_float64
0.27s call     test/test_ops.py::TestGradientsCUDA::test_forward_mode_AD_nn_functional_mse_loss_cuda_float64
```

## Conclusion

It seems that the the size of the sample inputs, at least for 

https://github.com/pytorch/pytorch/blob/607d720be154a44ddb5db20bee4999971cf967f5/torch/testing/_internal/common_methods_invocations.py#L753-L755

is fairly irrelevant, because it is drowned by the overhead of moving the tensors to CUDA.

Reply:
Cool; it used to be that gradchecks too MUCH longer with larger inputs, and those checks are still run on trunk (but not on Github CI), so we still might want to bias towards small sample inputs (for now, pending further review)

Reply:
Would you prefer 2. or 3.?

Reply:
L -> S or 2 or 3 SGTM

Reply:
Let's do 2.